hastexo/migrations/0010_add_user_foreign_key.py

```diff
@@ -1,10 +1,29 @@
 from django.conf import settings
-from django.db import migrations, models
+from django.db import migrations, models, OperationalError
+from django.db.models.base import ObjectDoesNotExist
 import django.db.models.deletion
 
 
 class Migration(migrations.Migration):
 
+    def check_stacks(apps, schema_editor):
+        Stack = apps.get_model("hastexo", "Stack")
+        AnonymousUserId = apps.get_model("student", "AnonymousUserId")
+        stacks_with_no_user = []
+        for stack in Stack.objects.all():
```

Comment:
Ok so this is a loop over _all_ `Stack` objects with a nested `get` on another object type. I worry that in an environment with thousands of stacks and hundreds of users this would take forever. I do appreciate that this is a migration so we might not worry about such things in general, but here I'm guessing we might actually hit standard 30-minute CI pipeline timeouts. So if this is the only way to do it then let's keep this, but if we _can_ make this faster, we should. 

Is there any chance this could be rewritten as a [filter query](https://docs.djangoproject.com/en/4.0/topics/db/queries/#retrieving-specific-objects-with-filters)?

Reply:
I believe we can use exclude, I've updated the patch just now, please have a look.